From a973d9520ff08dde57a7234da9abde827b3a9975 Mon Sep 17 00:00:00 2001
From: Daniel Barry <dbarry@vols.utk.edu>
Date: Tue, 7 Feb 2023 10:41:07 -0500
Subject: [PATCH] sysdetect: account for older CUDA versions

In CUDA 11.0 or greater, the macro "NVML_DEVICE_UUID_V2_BUFFER_SIZE" is defined.
Older versions of CUDA define "NVML_DEVICE_UUID_BUFFER_SIZE."

In order to support older versions of CUDA, these changes apply the appropriate macro.

These changes have been tested on the NVIDIA Tesla V100 architecture.
---
 src/components/sysdetect/nvidia_gpu.c | 4 ++--
 src/components/sysdetect/nvidia_gpu.h | 6 ++++++
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/components/sysdetect/nvidia_gpu.c b/src/components/sysdetect/nvidia_gpu.c
index 295590882..71f27d861 100644
--- a/src/components/sysdetect/nvidia_gpu.c
+++ b/src/components/sysdetect/nvidia_gpu.c
@@ -257,9 +257,9 @@ fill_dev_affinity_info( _sysdetect_gpu_info_u *info, int dev_count )
         NVML_CALL((*nvmlDeviceGetHandleByPciBusIdPtr)(bus_id_str, &device),
                   return);
 
-        char uuid_str[NVML_DEVICE_UUID_V2_BUFFER_SIZE] = { 0 };
+        char uuid_str[PAPI_NVML_DEV_BUFFER_SIZE] = { 0 };
         NVML_CALL((*nvmlDeviceGetUUIDPtr)(device, uuid_str,
-                                          NVML_DEVICE_UUID_V2_BUFFER_SIZE),
+                                          PAPI_NVML_DEV_BUFFER_SIZE),
                   return);
 
         _sysdetect_gpu_info_u *dev_info = &info[dev];
diff --git a/src/components/sysdetect/nvidia_gpu.h b/src/components/sysdetect/nvidia_gpu.h
index f4077f55b..96eb2575a 100644
--- a/src/components/sysdetect/nvidia_gpu.h
+++ b/src/components/sysdetect/nvidia_gpu.h
@@ -1,6 +1,12 @@
 #ifndef __NVIDIA_GPU_H__
 #define __NVIDIA_GPU_H__
 
+#if CUDA_VERSION >= 11000
+    #define PAPI_NVML_DEV_BUFFER_SIZE NVML_DEVICE_UUID_V2_BUFFER_SIZE
+#else
+    #define PAPI_NVML_DEV_BUFFER_SIZE NVML_DEVICE_UUID_BUFFER_SIZE
+#endif
+
 void open_nvidia_gpu_dev_type( _sysdetect_dev_type_info_t *dev_type_info );
 void close_nvidia_gpu_dev_type( _sysdetect_dev_type_info_t *dev_type_info );
 
-- 
2.39.0

